File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
            i = 0
            j = 0
            arr = []
            size = len(s)
            while i < size:
                while i < size and s[i] == ' ':
                    i += 1
                j = i
                while j < size and s[j] != ' ':
                    j += 1
                if i < size:
                    arr.append(s[i:j])
                i = j
            return ' '.join(arr[::-1])

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
            int i = 0, j = 0;
            size_t size = s.length();
            vector<string> arr;
            while(i < size)
            {
                while(i < size && s[i] == ' ')
                {
                    i++;
                }
                j = i;
                while(j < size && s[j] != ' ')
                {
                    j++;
                }
                if(i < size) 
                {
                    arr.push_back(s.substr(i,j-i));
                }
                i = j;
            }
            reverse(arr.begin(), arr.end());
            s = "";
            for(const string &str: arr)
            {
                s+=(str + ' ');
            }
            s.erase(s.size() - 1);
            return s;
    }
};